htetl/tasks/main.py

```diff
@@ -0,0 +1,150 @@
+import string
+import htetl.extract.emails
+import htetl.get_data as get_data
+import htetl.reverse_url as reverse_url
+import htetl.tasks.phones
+import htetl.util as util
+
+
+logger = logging.getLogger('luigi-interface')
+logger.setLevel(logging.DEBUG)
+
+
+class MakeGraph(luigi.Task):
+    """  reversing url task """
+
+    NEW_DATA_EXTRACTOR = luigi.BoolParameter(
+        config_path=dict(section='htetl-flags', name='new_data_extractor'),
+        default=False
+    )
+
+    def output(self):
+        out_path = 'data/graph.csv'
+        return luigi.LocalTarget(out_path)
+
+    def requires(self):
+        if self.NEW_DATA_EXTRACTOR:
+            return {
+                'email': htetl.extract.emails.ParseEmails(),
+                'phone': htetl.tasks.phones.ParsePhones(),
+                # TODO: This needs to be adapted to the new Page data
+                # extraction methods like ParseEmail and ParsePhones
+                'oid': get_data.RawPosterData(),
+            }
+        else:
+            return [get_data.RawPhoneData(),
+                    get_data.RawEmailData(),
+                    get_data.RawPosterData()]
+
+    def run(self):
+        """ data is small enough to use pandas for the processing """
+
+        out = []
+
+        # TODO: When htetl-flags:new_data_extractor is removed, remove the
+        # flatten call and clean up the for loop
+        for in_target in luigi.task.flatten(self.input()):
+            with in_target.open('r') as f:
+                data = pd.read_csv(f)
+
+            for i, (k, v) in enumerate(data.groupby(data.columns[-1])):
+                values = v.values.tolist()
+                page_ids = [x[0] for x in values]
+                offset_page_ids = page_ids[1:]
+                if len(page_ids) == 1:
+                    offset_page_ids = page_ids
+                else:
+                    offset_page_ids.append(page_ids[0])
+                out.append([
+                    (a, b)
+                    for a, b in zip(page_ids, offset_page_ids)
+                ])
+
+        out = [item for sublist in out for item in sublist]
+
+        G = nx.Graph()
+        G.add_edges_from(out)
+
+        sub_graphs = []
+        for i, x in enumerate(nx.connected_component_subgraphs(G)):
+            nodes = nx.nodes(x)
+            sub_graphs.append(list(zip([i] * len(nodes), nodes)))
+
+        sub_graphs = [item for sublist in sub_graphs for item in sublist]
+
+        df_out = pd.DataFrame(sub_graphs,
+                              columns=['entity_id',
+                                       'backpagepostid']).astype(int)
+
+        with self.output().open('w') as f:
+            df_out.to_csv(f, index=None)
+
+
+class LoadEntityIds(util.LoadPostgres):
+    table = 'backpageentities'
+
+    columns = [('entity_id', 'int'),
+               ('backpagepostid', 'int')]
+
+    header = True
+
+    def requires(self):
+        return MakeGraph()
+
+class ParseEmails(luigi.Task):
```

Comment:
how did ParseEmails end up in the main function and not its own file?

Reply:
I think this is the old `ParseEmails` function, not the newer one we recently created.

Reply:
So we're keeping this one until the new one works? If we're planning on removing it, then put a TODO here so that someone new looking at the code doesn't get confused by duplicated logic